README.md

```diff
@@ -1,53 +1,144 @@
 
-001_create_posts.rb
+A typical model file will contain code such as, but not limited to the following:
 
-we dont need to go too much into Migration syntax as they mostly learned this already or this readme can assume they are familiar with AR Basics such as migrations and stuff. but we'll have the migration code and link to migration docs for more information.
+* Custom scopes
+* Model instance methods
+* Default settings for database columns
+* Validations
+* Model-to-model relationships
+* Callbacks
+* Custom algorithms
 
-then we need a model file in app/models
+## Creating an ActiveRecord Model
 
-post.rb
+In order to get started, we will first create a RSpec test so that we can build out everything using a BDD approach. Assuming we already have RSpec installed, let's create a new directory and place our first spec in it: ```spec/models/post_spec.rb```. In that file place the following code:
 
-let's reiterate here convention over configuration, remind them of controller/action/view and now we have another one table/model/filename and migration_filename/migration_class
+```ruby
+require 'rails_helper'
 
-rake db:migrate (they know rake, and have done this)
+describe Post do
 
-accessing the model in rails console
+end
+```
 
-reading a model spec. (first_name/full_name is a good example). They've done this a ton
+If we run ```bundle exec rspec``` it will throw an error since we don't have any code in the application for our ```post``` model yet. To fix this create a new file in the ```app/models``` directory called ```post.rb``` and add the following code:
 
-explain models are separate from controllers and views - we connect a model to a controller and a view by loading the model in a controller action, no convention there...
+```ruby
+class Post
+end
+```
 
-database.yml defines are database connection and ActiveRecord::Base.connection holds those connections.
+This will get the tests passing, but it still has some weird errors, and this is because we need to create a schema file. You can do that by running ```rake db:migrate```, this will create the schema file and clear the warning. Now update the Post spec to test for a Post being created, it will look something like this:
 
-That's it for models in rails for now.
+```ruby
+describe Post do
+  it 'can be created' do
+    post = Post.create!(title: "My title", description: "The post description")
+    expect(post).to be_valid
+  end
+end
```

Comment:
It's the first time they are seeing all of this rails stuff. Shy away from also adding in writing tests. While our students are very good at reading tests, they aren't great at writing them by this time in the curriculum. We make sure they understand the content first, then a few units later after this we'll do a ton of TDD stuff. So let's just show them testing things out in `rails c` for right now